app/assets/javascripts/discourse/controllers/full-page-search.js.es6

```diff
@@ -90,7 +91,9 @@ export default Ember.Controller.extend({
     this._searchOnSortChange = false;
     if (term) {
       SortOrders.forEach(order => {
-        if (term.indexOf(order.term) > -1){
+        let term_start      = term.indexOf(order.term)
+        let char_after_term = term[term_start + (order.term || '').length]
+        if (term_start > -1 && (!char_after_term || char_after_term.match(/\s/))){
```

Comment:
Ah, oops, I realize the problem now, previously I misunderstood. Sorry about that.

In that case, I think it would be a lot clearer to use a regular expression rather than checking the characters manually like that. `/b` should match a boundary or end of string.

```
const matches = term.match(new RegExp(`${order.term}//b`);
if (matches) {
  // ... 
}
```